File: http/src/main/java/com/xinzy/http/DownloadCallback.java
package com.xinzy.http;

/**
 * Created by Xinzy on 2017/6/12.
 *
 */
public interface DownloadCallback {
    void onStart();
    void onLoading(long current, long total);
    void onEnd();
    void onFailure(SmartHttpException e);

    class DefaultDownloadCallback implements DownloadCallback {
        @Override
        public void onStart() { }

        @Override
        public void onLoading(long current, long total) { }

        @Override
        public void onEnd() { }

        @Override
        public void onFailure(SmartHttpException e) { }
    }
}

File: httpkotlin/src/main/java/com/xinzy/http/kotlin/DownloadCallback.kt
package com.xinzy.http.kotlin

/**
 * Created by Xinzy on 2018/10/12.
 */
interface DownloadCallback {
    fun onStart()
    fun onLoading(current: Long, total: Long)
    fun onEnd()
    fun onFailure(e: SmartHttpException)
}

class DefaultDownloadCallback: DownloadCallback {
    override fun onStart() {}

    override fun onLoading(current: Long, total: Long) {}

    override fun onEnd() {}

    override fun onFailure(e: SmartHttpException) {}
}